### PR TITLE
async-46: Rendering Guide and Code Comments

### PR DESCRIPTION
**Pull request by pwinston**
# Description
* New "guide" document on rendering.
* A pass through code comments in light of the guide and just cleaning them up.
* Remove some very old code we no longer use  at all.
    * From back when we created Octrees from single-scale images 
    * For testing Octree before we could even load multi-scale data.
## Type of change
- [x] New guide document
- [x] Revise code comments

### CODE REVIEW

**Comment by pwinston** [comment]
@sofroniewn it's somewhat in okay shape. It's better than nothing certainly. But I'd like to keep chipping away at it. I'm only putting in like an hour and here or there. So we can merge it as "rough docs" or wait and I'll post here when it's in better shape.

I still need to add more, but also fix up formatting and editing for clarity.

Also it's really kind of 20,000 foot view. Just describing the basic pieces and how they fit together. I'm not sure whether we should dive deeper in the "guide" or defer to improving code comments instead at some point.

**Comment by sofroniewn** [comment]
We're still working on the final 2 code PRs for the release, we decided to push for one extra feature. They will almost certainly need till tomorrow, and maybe Saturday. So I'll wait until they are in to review this. Feel free to keep chipping in the mean time, and thanks for the update.

**Comment by pwinston** [comment]
Sounds good. 

As for "guide" vs. "code comments", I feel the guide is something you want to read straight through and get a good overview. And it will duplicate some information that's in comments. And that's okay, because it's nice to have it gathered in one place. And read it in one shot. But that ultimately there should be a lot of details in class and method comments that are not part of the guide. 

Another way to say that, I'm trying to make the guide a longer lasting document that hopefully does not go out of date as fast as code comments about a specific class or method might.

So I don't know if the definition of our "guides" has been discussed. But that was my assumption as far as the intent. The perfmon guide was different in that it's really more of tutorial. There's no real tutorial for async rendering.

**Comment by sofroniewn** [comment]
That sounds good to me

**Comment by pwinston** [comment]
@sofroniewn I was confused why some of my RST links in the guide doc work worked and others didn't. Finally I realized we don't even generate docs for private modules or files. This means there are no API docs for the tiled visual files under `_vispy`. And while `napari.experimental.chunk` has no underscore all of the files in there do:

```
__init__.py
_cache.py
_commands
_delay_queue.py
_info.py
_loader.py
_pool.py
_pool_group.py
_request.py
_utils.py
```

I can see why we skip private files. None of this is in the API and we don't want to advertise it as being in the API. Yet for this guide doc, it seems weird you can't link to most of the code. I think the question is should there be "rst generated docs" for non-api files at some point?

For now I'll just keep making the links. RST handles links to "missing" stuff fine. It gets styled as code, but you can't click on it.

**Comment by sofroniewn** [comment]
> For now I'll just keep making the links. RST handles links to "missing" stuff fine. It gets styled as code, but you can't click on it.

Yeah, that makes sense for now. We can always revisit/ have a version with all private things built too.

File: docs/source/events/rendering.rst
Comment (sofroniewn) [inline docs/source/events/rendering.rst:381]: I can't remember if I saw this elsewhere earlier, or if it's here and i'm just missing it, but should we add a section here like `Known Octree Limitations` and talk about how contrast limits, opacity, colormaps, and different blending modes aren't supported yet. Also any known dtypes that aren't supported. I haven't tested with multiscale timeseries, but if you know that doesn't work could add that too.

**Review by sofroniewn — COMMENTED** [review]
Overall looking very good. I left one comment about adding a section on "known octree limitations" (or similar) if you don't have one already.

File: docs/source/events/rendering.rst
Comment (pwinston) [reply docs/source/events/rendering.rst:381]: I put in several new sections for each of these. Giving my assessment of where we might want to go. Although a fresh set of ideas might have solutions I'm not thinking of.

**Comment by pwinston** [comment]
I've marked this ready for review. I could keep polishing it, but a lot of details will still change as the code evolves, so I don't think this is permanent documentation "for the ages" yet.

I feel in this state it serves the purpose of being a good overview of the whole state of things. Basically, I think this guide is a good step 0 if you are going to dive into the code.

Open to suggestions though, if any section needs more work or is missing let me know.

Also, I made a pass through all the code comments. Those also could be improved, but I think they are in a reasonable state now.

**Review by sofroniewn — APPROVED** [review]
Ok, this looks great! Thanks for adding that info on the clims/ blending and in general the future work section. That will be very helpful. I will merge now!!

**Pull request merged**
